src/sentry/incidents/serializers/alert_rule.py

```diff
@@ -513,12 +513,24 @@ def create(self, validated_data):
                 "organizations:workflow-engine-metric-alert-dual-write", alert_rule.organization
             )
             if should_dual_write:
-                migrate_alert_rule(alert_rule, user)
+                try:
+                    migrate_alert_rule(alert_rule, user)
+                except Exception as e:
+                    logger.exception(
+                        "Error when dual writing alert rule", extra={"details": str(e)}
+                    )
+                    raise BadRequest
```

Comment:
it might be nice if the exception included why it failed, then proxy that through the BadRequest so the API can have more specific errors.